streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -98,4 +116,47 @@ public String toString() {
         }
         return sb.toString();
     }
+
+    protected class NodeMetrics  {
+        final StreamsMetrics metrics;
+        final String metricGrpName;
+        final Map<String, String> metricTags;
+
+        final Sensor nodeProcessTimeSensor;
+        final Sensor nodePunctuateTimeSensor;
+        final Sensor contextForwardSensor;
+        final Sensor nodeTaskCreationSensor;
+        final Sensor nodeTaskDestructionSensor;
+
+
+        public NodeMetrics(StreamsMetrics metrics, String name, String sensorNamePrefix) {
+            String scope = "node";
+            this.metrics = metrics;
+            this.metricGrpName = "node-metrics-" + name;
+            this.metricTags = new LinkedHashMap<>();
+            this.metricTags.put("node-id", "-" + name);
+
+            // these are all latency metrics
+            this.nodeProcessTimeSensor = metrics.addLatencySensor(scope, name, "process", Sensor.RecordLevel.SENSOR_DEBUG);
+            this.nodePunctuateTimeSensor = metrics.addLatencySensor(scope, name, "punctuate", Sensor.RecordLevel.SENSOR_DEBUG);
+
+            this.contextForwardSensor = metrics.sensor(sensorNamePrefix + "node-forward-time" + name, Sensor.RecordLevel.SENSOR_DEBUG);
```

Comment:
@aartigupta @enothereska That's right, but shouldn't the `sensorNamePrefix` contains that? Currently it only contains the taskId but I think it should include both since each task can have its own copy of the topology. As for the group name, it is used for grouping metrics that may come from different classes even (see producer's `Sender` and `Selector` classes), so it should not include the node id, but rather just "streams-processor-node-metrics".